models/surveys.go

```diff
@@ -729,3 +887,30 @@ func createStmt(sqlStatement string, db *sql.DB) (*sql.Stmt, error) {
 func (api *API) Close() {
 	api.AllSurveysStmt.Close()
 }
+
+// Roll back a given transaction and log any errors which occur
+func rollBack(tx *sql.Tx) {
+	err := tx.Rollback()
+	if err != nil {
+		logError("Error rolling back database transaction", err)
+	}
+}
+
+// Log and message and an error and send a 500 HTTP response
+func logErrorAndRespond500(w http.ResponseWriter, logMessage string, err error) {
```

Comment:
feels like this should a generic logError function where we pass in the http status code we want to return